### PR TITLE
Store and retrieve transaction results by EN RPC

### PR DESCRIPTION
**Pull request by m4ksio**
EN part of #2100

As this is intended to deploy mid-spork I added a new index (block_id, tx_index) rather than modify existing one. 
For a new spork we should probably combine indexes and use badger traversal to retrieve data needed and avoid duplicating tx results (although they are rather small.

Notice, this will only allow of retrieval by index for new data, which has been executed after this change is live. Previously executed data won't be available but rebuilding index for past data is a separate issue.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -106,3 +106,5 @@ require (
 )
 
 replace mellium.im/sasl => github.com/mellium/sasl v0.2.1
+
+//replace github.com/onflow/flow => ../flow
```
Comment (koko1123) [inline go.mod:110]: why aren't we using the local version?

File: storage/badger/transaction_results.go [hunk 2]
```diff
@@ -13,14 +15,21 @@ import (
...
+	db         *badger.DB
+	cache      *Cache
+	indexCache *Cache
 }
 
 func KeyFromBlockIDTransactionID(blockID flow.Identifier, txID flow.Identifier) string {
 	return fmt.Sprintf("%x%x", blockID, txID)
 }
 
+func KeyFromBlockIDIndex(blockID flow.Identifier, txIndex uint32) string {
```
Comment (koko1123) [inline storage/badger/transaction_results.go:27]: smart!

**Review by koko1123 — APPROVED** [review]
Great work on such short notice!

File: storage/mocks/storage.go [hunk 3]
```diff
@@ -600,6 +600,21 @@ func (mr *MockEventsMockRecorder) ByBlockIDTransactionID(arg0, arg1 interface{})
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "ByBlockIDTransactionID", reflect.TypeOf((*MockEvents)(nil).ByBlockIDTransactionID), arg0, arg1)
 }
 
+// ByBlockIDTransactionIndex mocks base method
```
Comment (ramtinms) [inline storage/mocks/storage.go:603]: any reason for having to mocks files on under mocks and one under mock ?

File: go.mod [hunk 4]
```diff
@@ -59,13 +59,13 @@ require (
...
 	github.com/onflow/cadence v0.23.2
-	github.com/onflow/flow v0.2.3-0.20220131193101-d4e2ca43a621
+	github.com/onflow/flow v0.2.3-0.20220315185722-b1c4393db996
 	github.com/onflow/flow-core-contracts/lib/go/contracts v0.10.2-0.20220303222312-6b9de4f3fea5
 	github.com/onflow/flow-core-contracts/lib/go/templates v0.10.1
 	github.com/onflow/flow-emulator v0.20.3
 	github.com/onflow/flow-go-sdk v0.24.0
 	github.com/onflow/flow-go/crypto v0.24.3
-	github.com/onflow/flow/protobuf/go/flow v0.2.4-0.20220304041411-6d91cd04a33a
+	github.com/onflow/flow/protobuf/go/flow v0.2.4-0.20220315185722-b1c4393db996
```
Comment (ramtinms) [inline go.mod:68]: do we need to release a new version there ?

**Review by ramtinms — APPROVED** [review]
Looks good to me, just put two minor comments

File: go.mod (hunk 4 above)
Comment (m4ksio) [reply go.mod:68]: Yes, once we merge the changes from other repo

File: storage/mocks/storage.go (hunk 3 above)
Comment (m4ksio) [reply storage/mocks/storage.go:603]: Two different mocking libraries maybe?

**Pull request merged**